### PR TITLE
Update api.js

### PR DESCRIPTION
**Pull request by drndos**
Fix issue where crowd server sends content type "application/json;charset=UTF-8"

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default class CrowdApi {
           if (path === '/group/membership' && response.statusCode === 200 && responseData) {
             // Return the raw XML response for /group/membership, since it doesn't support JSON.
             resolve(responseData);
-          } else if (response.headers['content-type'] !== 'application/json') {
+          } else if (!response.headers['content-type'].startsWith("application/json")) {
```
Comment (reblace) [inline src/api.js]: This will crash if there's no content-type header. Should start it with a null check.

**Comment by drndos** [comment]
Fixed possible NPE

File: src/api.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ export default class CrowdApi {
           if (path === '/group/membership' && response.statusCode === 200 && responseData) {
             // Return the raw XML response for /group/membership, since it doesn't support JSON.
             resolve(responseData);
-          } else if (response.headers['content-type'] !== 'application/json') {
+          } else if (!response.headers['content-type'] || !response.headers['content-type'].startsWith("application/json")) {
```
Comment (ghengeveld) [inline src/api.js]: Please do a split on `;`, then check against the first item.

```js
response.headers['content-type'].split(';')[0] !== 'application/json'
```

**Comment by drndos** [comment]
Fixed to better check compliance with rfc1341

**Comment by ghengeveld** [comment]
Excellent, thanks!

**Comment by ghengeveld** [comment]
Released in v1.4.2. Thanks for contributing!

**Pull request merged**